slice-core/src/main/java/com/cognifide/slice/core/internal/monitoring/webconsole/StatisticsHtmlTablesRenderer.java

```diff
@@ -0,0 +1,197 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.slice.core.internal.monitoring.webconsole;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.cognifide.slice.core.internal.monitoring.ModelUsageData;
+
+public class StatisticsHtmlTablesRenderer {
+
+	private Map<String, ModelUsageData> injectorsStatistics;
+
+	public StatisticsHtmlTablesRenderer(Map<String, ModelUsageData> injectorsStatistics) {
+		this.injectorsStatistics = injectorsStatistics;
+	}
+
+	@Override
+	public String toString() {
+		StringBuilder tableBuilder = new StringBuilder();
+		tableBuilder.append("<style type='text/css'>" + getResourceAsString("jquery.treetable.css") + "</style>");
+		for (Entry<String, ModelUsageData> injectorEntry : injectorsStatistics.entrySet()) {
+			String header = injectorEntry.getKey();
+			String id = header.replaceAll(" > ", "-").replaceAll(" ", "_");
+			renderFlatTable(tableBuilder, id, header, injectorEntry.getValue());
+			renderTreeTable(tableBuilder, id, header, injectorEntry.getValue());
+			tableBuilder.append("<hr style='margin-bottom: 20px'>");
+		}
+		tableBuilder.append("<hr><hr>");
+		tableBuilder.append(
+				"<form action='' method='get'><input type='hidden' name='reset'><button type='submit'>RESET</button></form>");
+		// remove root children indentation
+		tableBuilder.append(
+				"<script>$(document).ready(function() {$(\"tr\").find(\"td:first\").find(\"span[style='padding-left: 0px;']\").remove();});</script>");
+		tableBuilder
+				.append("<script type='text/javascript'>" + getResourceAsString("jquery.treetable.js") + "</script>");
+		return tableBuilder.toString();
+	}
+
+	private void renderFlatTable(StringBuilder tableBuilder, String id, String header, ModelUsageData treeItem) {
+
+		tableBuilder.append("<table id='summary-" + id + "' class='tablesorter nicetable noauto'>");
+		tableBuilder.append(renderCaption(id, header, true));
+		tableBuilder.append(
+				"<thead><tr><th>Class</th><th>Instances Injected</th><th>Total Init Time [ms]</th class=\"{sorter: 'floating'}\"><th>Avg. Init Time [ms]</th></tr></thead><tbody>");
+		for (Entry<Class<?>, ModelUsageData> entry : flattenStats(treeItem).entrySet()) {
+			ModelUsageData stats = entry.getValue();
+			Long instancesCount = stats.getCount();
+			Long totalTime = stats.getTotalTime();
+			double avgTime = stats.getAverageTime();
+			tableBuilder.append("<tr><td>");
+			tableBuilder.append(entry.getKey().getName());
+			tableBuilder.append("</td><td>");
+			tableBuilder.append(instancesCount.toString());
+			tableBuilder.append("</td><td>");
+			tableBuilder.append(totalTime.toString());
+			tableBuilder.append("</td><td>");

```

Comment:
How about creating a method to render a single cell instead of calling this time and time again?